### DISCUSSION TITLE
Deploying frameworkless serverless only functions

### DISCUSSION THREAD

**Discussion by ahsath** [vercel/vercel#4394] [Help] [answered]
Hi folks,

I wanted to build an API using graphql for an static ecommerce website built with Stenciljs but only use the serverless service with no framework or build step at all but still having the nice git integration.

Ideally my thoughs are having 3 separate repos:

1. For the static website under `example.com`

1. For the API itself under `api.example.com/api/graphql`

1. For a SPA under `admin.example.com`

**So my questions are:**
 
1. Can I deploy frameworkless serverless only functions?

1. Can I use the same custom domain on different git integrations/projects?

**Comment by mcsdevv** [comment, marked as answer]
Hello,

You can deploy projects consisting of only Serverless Functions, can you tell us more about your use case though?

Yep, you can use the same custom domain on multiple projects!

**Reply by ahsath** [reply]
Thank you for replaying,

My use case regarding serverless functions is to have a GraphQL API written with apollo server (though I haven't dig deeper on how to do it) so it can be consumed by both Stenciljs when building the static site(dev/prod) and the admin dashboard as a SPA for CRUD operations

**Reply by mcsdevv** [reply]
Thanks for letting me know! I'd strongly recommend using a third-party service for the GraphQL API, there's loads of great options available. You might run into some limitations doing this with Serverless Functions but technically can be done.

**Reply by cdock1029** [reply]
@ahsath Have you tried https://hasura.io/ ? It really makes it easy to have a GraphQL service up and running without worrying about wiring up the api connection to the database, at least to get you started with a prototype. Handles efficiently translating GraphQL requests to SQL queries that is hard to get right manually. Easy to host on Heroku and such.

**Reply by ahsath** [reply]
I've had look into it, but hasura uses postgres as database and I would like to use MongoDB with Atlas
